vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1003,6 +1003,10 @@ func resourceVSphereVirtualMachineCustomizeDiff(d *schema.ResourceDiff, meta int
 			}
 			fallthrough
 		default:
+			// If guest_id is not set and the source is not a Content Library item, set it to the default.
+			if d.Get("guest_id") == "" {
+				d.SetNew("guest_id", "other-64")
+			}
```

Comment:
Why not use a DefaultFunc in the schema instead?

Reply:
I didn't think I could use `DefaultFunc` with `Computed` since `Default` can't be used. I was mistaken though. That is a better route! Thanks for pointing it out.

Reply:
Turns out `DefaultFunc` isn't supposed to be used with `Computed`, even though it does work. We've got to stick with this being computed though for things like Content Library where there is no way to read the values before deploying.